packages/mermaid/src/rendering-util/handle-markdown-text.ts

```diff
@@ -58,10 +63,15 @@ export function markdownToLines(markdown: string): MarkdownLine[] {
 
 export function markdownToHTML(markdown: string) {
   const { children } = fromMarkdown(markdown);
+  const markdownAutoWrap = getConfig().markdownAutoWrap;
 
   function output(node: Content): string {
     if (node.type === 'text') {
-      return node.value.replace(/\n/g, '<br/>');
+      if (markdownAutoWrap === false) {
+        return node.value.replace(/\n/g, '<br/>').replace(/ /g, '&nbsp;');
+      } else {
+        return node.value.replace(/\n/g, '<br/>');
+      }
```

Comment:
nit: You don't need this `else`, as the `if` ends in return. 
```suggestion
      }
      return node.value.replace(/\n/g, '<br/>');
```